certbot/tests/util_test.py

```diff
@@ -145,65 +143,29 @@ def setUp(self):
         self.path = os.path.join(self.tempdir, "foo")
         filesystem.mkdir(self.path, 0o600)
 
-        self.uid = misc.os_geteuid()
-
     def _call(self, directory, mode):
         from certbot.util import make_or_verify_dir
-        return make_or_verify_dir(directory, mode, self.uid, strict=True)
+        return make_or_verify_dir(directory, mode, strict=True)
 
     def test_creates_dir_when_missing(self):
         path = os.path.join(self.tempdir, "bar")
         self._call(path, 0o650)
         self.assertTrue(os.path.isdir(path))
-        self.assertTrue(misc.compare_file_modes(os.stat(path).st_mode, 0o650))
+        self.assertTrue(filesystem.check_mode(path, 0o650))
 
     def test_existing_correct_mode_does_not_fail(self):
         self._call(self.path, 0o600)
-        self.assertTrue(misc.compare_file_modes(os.stat(self.path).st_mode, 0o600))
+        self.assertTrue(filesystem.check_mode(self.path, 0o600))
 
-    @test_util.skip_on_windows('Umask modes are mostly ignored on Windows.')
     def test_existing_wrong_mode_fails(self):
-        self.assertRaises(errors.Error, self._call, self.path, 0o400)
+        self.assertRaises(errors.Error, self._call, self.path, 0o644)
 
     def test_reraises_os_error(self):
         with mock.patch.object(filesystem, "makedirs") as makedirs:
             makedirs.side_effect = OSError()
             self.assertRaises(OSError, self._call, "bar", 12312312)
 
 
-class CheckPermissionsTest(test_util.TempDirTestCase):
```

Comment:
We probably want a unit test for the new methods.

Reply:
Already done in #6964. This test suite is in fact an limited and old version used for `util_test.check_permissions` function that is removed by this PR.

Reply:
I did actually mean unit tests for `check_owner`/`check_mode`/`check_permissions`, which it doesn't look like there are unit tests for. But if I'm correct, these are only used in testing code anyway. I see they reuse internal code from the `filesystem.py` module, which I suspect is why they're there in the first place. I don't love the dissonance of testing-only code in the same module as production code; some other options include adding unit tests for them, or placing them somewhere else. What do you think?

Reply:
No they are used in `safe_open`. So some unit tests then.

Reply:
I added the tests.